gdal/frmts/rda/rdadataset.cpp

```diff
@@ -656,6 +704,50 @@ bool GDALRDADataset::ParseConnectionString( GDALOpenInfo* poOpenInfo )
             json_object_get_boolean(poDeleteOnClose));
     }
 
+    json_object* poMaxConnections = json_ex_get_object_by_path(poObj,
+                                                              "options.max-connections");
+    if( poMaxConnections &&
+        json_object_get_type(poMaxConnections) == json_type_int )
+    {
+        MaxCurlConnectionsSet(json_object_get_int(poMaxConnections));
+    }
+
+    json_object* poEnforceAdviseRead = json_ex_get_object_by_path(poObj,
+                                                               "options.advise-read");
+    if( poEnforceAdviseRead &&
+        json_object_get_type(poEnforceAdviseRead) == json_type_boolean )
+    {
+        m_bAdviseRead = CPL_TO_BOOL(
+                json_object_get_boolean(poEnforceAdviseRead));
+    }
+
+    if(m_osType == RDADatasetType::TEMPLATE)
+    {
+        json_object *poParams = json_object_object_get(poObj, "params");
+
+        if(poParams != nullptr) {
+            array_list *res = json_object_get_array(poParams);
+            if(res != nullptr) {
+                for (decltype(res->length) i = 0; i < res->length; i++) {
+                    json_object *ds = (json_object *) array_list_get_idx(res, i);
+                    if (ds != nullptr) {
+                        json_object_iter it;
+                        it.key = nullptr;
+                        it.val = nullptr;
+                        it.entry = nullptr;
+                        json_object_object_foreachC( ds, it )
+                        {
+                            CPLString tkey = it.key;
+                            CPLString tval = json_object_get_string(it.val);
```

Comment:
json_object_get_string() could return nullptr, and thus cause affectation to CPLString to segfault

Reply:
explicitly check it.key, it.val and the return from json_object_get_string for nullptr